core/templates/services/png-sanitizer.service.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview PNG related checks and sanitization service.
+ */
+
+import { Injectable } from '@angular/core';
+import { DomSanitizer, SafeResourceUrl } from '@angular/platform-browser';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PngSanitizerService {
```

Comment:
After migrating,
we use [src] instead of ng-src,
Since the image data is encoded in base64 format Angular is throwing errors as the URL is unsafe. Therefore it results in not rendering images in Dom.
![Screenshot from 2021-03-31 11-44-26](https://user-images.githubusercontent.com/54100702/113832399-384f0480-97a6-11eb-8b28-6785c20c2098.png)

So, I've implemented this sanitization helper service which is similar to the SVG sanitization service we already have.
I'm slightly skeptical about this method.
PTAL @seanlip @srijanreddy98

Reply:
Hm, I think it seems reasonable, but I would like @srijanreddy98 to confirm that we actually do need this class in the first place. It would be better not to need it, if possible, but I don't know enough about what is possible natively in Angular, so it would be great if he could weigh in.

For the specific validation, can you please get a review from @kevintab95, who I believe wrote the similar validation for SVG? @kevintab95 is looking after images in general so he would be a good person to advise here.

Thanks!

Reply:
PTAL @srijanreddy98 @kevintab95

Reply:
The base64 validation is fine, I think. But it seems to share some code with the SVG sanitizer, can we create a more general method for all supported base64 images strings?

Reply:
@Radesh-kumar I don't think this bypass is required.

> Since the image data is encoded in base64 format Angular is throwing errors as the URL is unsafe

I don't think this is true. Find out what exactly is tripping up the angular security. Look into angulars source code. They use a regex for this. Check why is our data not passing the regex.

Here is also something that might help you:
https://en.wikipedia.org/wiki/Base64

There is a table. With allowed charset. Just an FYI, Notice that there is no `%` in the list.

Reply:
Actually in the encoded base64 data `%` symbol exists. I think it might be causing this error. I've tried to convert base64 data URI  to blob,
![Screenshot from 2021-04-14 09-52-18](https://user-images.githubusercontent.com/54100702/114654145-2e2d8880-9d07-11eb-86d0-555b833633b8.png)
It seems there is a problem with encoding.
I got this error.
![Screenshot from 2021-04-14 09-43-16](https://user-images.githubusercontent.com/54100702/114654210-5321fb80-9d07-11eb-828c-5d9b2602c8d2.png)
How should I proceed?
PTAL @srijanreddy98

Reply:
Thanks, @srijanreddy98 this issue has been resolved.